### PR TITLE
Make `eth_gasPrice` aware of the base fee market

### PR DESCRIPTION
**Pull request by fab-10**
## PR description

The `eth_gasPrice` is not taking in account the base fee market values, when the network supports them, so it is possible that it returns values that are not good for pricing a tx for inclusion in the next block.
Specifically the calculation of gas price lower bound only rely on the runtime value of `minGasPrice`, but on base fee market networks, it could be that `baseFee` for the next block is higher than the `minGasPrice`, for example for devnets using the default, the `minGasPrice=1kwei` and the genesis sets `baseFee=1gwei` and the result is that senders thinks the can just price the gas at 1kwei to the the tx included, which is not the case.

This PR takes in consideration the baseFee value when calculating the lower bound, along with the fix I did some refactoring to move the gas price calculation into `BlockchainQueries` and continue to directly use `MiningParameters` to get the gas market info.

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java [hunk 1]
```diff
@@ -163,104 +192,142 @@ public void shouldReturnLimitedPriceWhenUpperBoundIsPresent() {
...
+    long upperBoundCoefficient = 200;
+    verifyGasPriceLimit(lowerBoundCoefficient, upperBoundCoefficient, gasPrice);
   }
 
+  //  @Test
+  //  public void shouldReturnConfiguredGasPriceLowerBoundAtGenesis() {
+  //    mockBaseFeeMarket();
+  //    mockBlockchain(0, 0);
+  //
+  //  }
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java:206]: remove if not needed

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java [hunk 2]
```diff
@@ -163,104 +192,142 @@ public void shouldReturnLimitedPriceWhenUpperBoundIsPresent() {
...
             request.getRequest().getId(), Wei.of(expectedGasPrice).toShortHexString());
 
-    when(blockchain.getChainHeadBlockNumber()).thenReturn(10L);
-    when(blockchain.getBlockByNumber(anyLong()))
-        .thenAnswer(invocation -> createFakeBlock(invocation.getArgument(0, Long.class)));
+    final var chainHeadBlockNumber = 10L;
+    mockBlockchain(chainHeadBlockNumber, 1);
+
+    // baseFee for next block = 0x7c61 (31_841)
+    // BlockchainQueries returns gasPrice = 0x5b8d80 (60_000)
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java:237]: remove if not needed

**Review by macfarla — COMMENTED** [review]
some commented out code hanging around, overall looks good

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java (hunk 1 above)
Comment (fab-10) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java]: done

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java (hunk 2 above)
Comment (fab-10) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthGasPriceTest.java]: done

**Review by macfarla — APPROVED** [review]

**Pull request merged**